lib/AST/Decl.cpp

```diff
@@ -2009,6 +2010,20 @@ VarDecl *VarDecl::getDefinition(ASTContext &C) {
   return nullptr;
 }
 
+// Checked C bounds information
+
+bool VarDecl::hasBoundsExpr() const {
+  return BoundsExpr != nullptr;
+}
+
+Expr *VarDecl::getBoundsExpr() {
+  return BoundsExpr;
+}
+
+void VarDecl::setBoundsExpr(Expr *E) {
```

Comment:
I'm not sure if it's a good example to follow, but the code below for setInit() first tries to deallocate any existing stored expression ('BoundsExpr') in this case before storing the pointer.  Do we need to do anything to make sure that the lifetime of 'E' is appropriate for the lifetime of 'this'?